docs/discussion/Multi-Tenancy Support.md

```diff
@@ -0,0 +1,285 @@
+
+#### Cache
+In the current configuration, Ratify offers support for both in-memory cache and distributed cache.
+
+Concerning in-memory cache, neither the cluster admin nor the team admin possesses direct access. The key focus here is on Ratify's ability to effectively partition the cache for distinct teams, mitigating concerns regarding cache isolation.
+
+In the case of distributed cache, exemplified by Redis, team admins and developers lack direct access to the Redis cluster unless explicitly granted permissions. Notably, cluster admins retain the capability to access Redis pods for manual data retrieval or updates—an occurrence often encountered during debugging tasks. It is crucial to acknowledge a specific scenario where granting a team admin access to Redis pods may inadvertently expose data from other teams within those pods, warranting careful consideration.
+
+##### User Scenario
+1. The in-memory cache is inaccessible to both the team admin and the cluster admin, even though they can access the Ratify pods.
+2. The cluster admin can read and update the data in the distributed cache, such as Redis.
+3. The team admin cannot access the data in the distributed cache without the proper permission.
+4. Once the team admin is granted the permission to the Redis cluster and the secret for the encryption key, they can access the data in the distributed cache.
+5. Different teams can store data with the same key in the distributed cache without any conflict.
+6. The scope of the data that the team admin can access depends on how the Redis cluster is deployed. If there is a single Redis cluster for all the namespaces, the team admin can read data from other namespaces. If there is a separate Redis cluster for each namespace, the team admin can only access data in their own namespace.
+
+##### Cache Partition
+1. If each namespace has its own Redis instance, there is no need for cache partitioning, because the data is already separated by namespace.
+2. If there is a single Redis cluster for the whole cluster, cache partitioning is necessary, because different teams might have data with the same key. In this case, we can use the namespace as a prefix for the key to avoid conflict.
+
+##### Redis Deployment and Ratify Deployment
+The deployment strategy affects the Cache Isolation, as shown above.
+
+Ratify can be deployed either per namespace or per cluster. Redis can also be deployed either per namespace or per cluster. The table below shows the different deployment combinations and their approaches to cache isolation.
+
+|  | Ratify per Namespace | Ratify per Cluster |
+|--|--|--|
+| Redis per Namespace | Data Isolated | It requires Ratify route requests to Redis instances in different namespaces |
+| Redis per Cluster | Prefix keys and restrict access via [ACL](https://redis.io/docs/management/security/acl/) | Prefix keys and restrict access via [ACL](https://redis.io/docs/management/security/acl/) |
```

Comment:
I think in the intial implementation , Redis per Cluster is okay. In daily workflow, it is rare for customers to view data in the redis cache. We could just start with ONLY cluster admin having access.